### ISSUE TITLE
BinarizedMNIST can't load full training set

### ISSUE THREAD

**Issue by goodfeli** [lisa-lab/pylearn2#1397]
The BinarizedMNIST class doesn't offer any interface for loading the full 60,000 example training set, only the 50,000 example sub-train set used for hyperparameter validation. This means you can't really use it to reproduce state of the art results on the test set that were obtained by training on the full training set. It's also not consistent with the original MNIST class.

@vdumoulin wrote /  @lamblin reviewed this, let me know if that's the intended behavior for some reason.

**Comment by goodfeli** [comment]
Likewise, it appears not to have any labels, and I'm pretty sure Russ's DBM results involved training an RBM over h2 and y followed by PCD on the joint of v, h1, h2, and y.

**Comment by vdumoulin** [comment]
It was meant to be a wrapper around the unlabeled binarized MNIST data that's been used for Russ' DBM as well as NADE and VAE, hosted at `http://www.cs.toronto.edu/~larocheh/public/datasets/binarized_mnist/binarized_mnist_{train,valid,test}.amat`.

This is the only version of that exact binarization which I have access to, so this is why labels aren't available. If anyone knows where a version including labels can be downloaded, we can update the download script and the class to make labels available.

There is no reason not to allow for training on the whole training set other than the dataset was constructed to be a wrapper around the three previously-mentioned files, so that's definitely something that can be implemented.
